src/esp/gfx/Viewer.h

```diff
@@ -84,6 +84,15 @@ class Viewer : public Magnum::Platform::Application {
   std::vector<int> objectIDs_;
 
   Magnum::Timeline timeline_;
+
+  const Eigen::IOFormat eigenFlatFormat_{Eigen::StreamPrecision,
+                                         Eigen::DontAlignCols,
+                                         ", ",  // coef separator
+                                         ", ",  // row separator
+                                         "",    // row prefix
+                                         "",    // col prefix
+                                         "[",   // mat prefix
+                                         "]"};  // mat suffix
```

Comment:
I encountered [this in core/esp.h](https://github.com/facebookresearch/habitat-sim/blob/1d4bd7bc8c73f6f4674e6e2c553930eff1a0d606/src/esp/core/esp.h#L32-L41) when going over the code few months ago and assumed all code was using it already:

```cpp
//! Eigen JSON string format specification
static const IOFormat
    kJsonFormat(StreamPrecision, DontAlignCols, ",", ",", "", "", "[", "]");

//! Write Eigen matrix types into ostream in JSON string format
template <typename T, int numRows, int numCols>
std::ostream& operator<<(std::ostream& os,
                         const Matrix<T, numRows, numCols>& matrix) {
  return os << matrix.format(kJsonFormat);
}
```

For the things printed in viewer, are those not `Matrix` instances and thus it doesn't work? Are those `Eigen::Array`s? Or some `Eigen::Map<Something, Something>`? What about adding another `operator<<` overload there instead so you don't need to specify `.format()` every time?

Reply:
Thanks for catching this @mosra -- indeed, an `operator<<` overload in core/esp.h ought to be able to handle formatting without the need to explicitly call `format()` every time.

Reply:
Good to know the similar format was actually defined in the esp.h.
Will take a look, and see if it can be used. Thanks.